File: solutions/.py/binary-search.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        
        a = 0
        b = len(nums) - 1
        c = (a + b) // 2
        while a <= b:
            if nums[c] == target:
                return c
            elif nums[c] < target:
                a = c + 1
                c = (a + b) // 2
            elif nums[c] > target:
                b = c - 1
                c = (a + b) // 2
        return -1

File: solutions/.java/binary-search.java
class Solution {
    public int search(int[] nums, int target) {
        int a = 0;
        int c = nums.length - 1;
        int b = (a + c) / 2;

        while(a <= c){
            if(nums[b] == target){
                return b;
            }else if(nums[b] < target){
                a = b + 1;
                b = (a + c) / 2;
            }else{
                c = b - 1;
                b = (a + c) / 2;
            }
        }
        return -1;
    }
}